src/phoenix/datasets/validation.py

```diff
@@ -56,60 +59,89 @@ def _check_valid_embedding_data(dataframe: DataFrame, schema: Schema) -> List[er
 
     embedding_errors: List[err.ValidationError] = []
     for embedding_name, column_names in embedding_col_names.items():
-        vector_length = None
         vector_column = dataframe[column_names.vector_column_name]
+        embedding_errors += _validate_embedding_vector(
+            embedding_name, vector_column, column_names.vector_column_name
+        )
+
+    return embedding_errors
+
+
+def _check_valid_prompt_response_data(
+    dataframe: DataFrame, schema: Schema
+) -> List[err.ValidationError]:
+    prompt_response_errors: List[err.ValidationError] = []
+
+    prompt_response_column_names = {
+        "prompt": schema.prompt_column_names,
+        "response": schema.response_column_names,
+    }
+    for name, column_names in prompt_response_column_names.items():
+        if column_names is not None:
+            vector_column = dataframe[column_names.vector_column_name]
+            prompt_response_errors += _validate_embedding_vector(
```

Comment:
did we check somewhere that `raw_data_column_name` is not None for the `prompt=/response=` pair?

Reply:
Do we want to enforce that? I agree that is kinda missing the point to send just the vectors, and since the `prompt`/`response` pairs are special case of embedding features, I am okay enforcing it

Reply:
Will not be adding a check returning an error if `raw_data_column_name` is not sent. Possibly to be given feedback in UI